code/core/src/manager/components/useLocationHash.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+
+const hashMonitor = {
+  currentHash: globalThis.window?.location.hash ?? '',
+  intervalId: null as ReturnType<typeof setInterval> | null,
+  listeners: new Set<(hash: string) => void>(),
```

Comment:
What is this used for?

Why is subscribing to a polling `interval` on `window.location.hash` the right approach for that?

Reply:
Because due to how our routing works, we can't reliably use a pushstate listener.
I invite you to find a working solution that doesn't involve polling. I gave up after wasting half a day on it.
The solution I built at least shares a single poller across multiple callsites.